build/elasticsearch/bin/es-docker

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+# Run Elasticsearch and allow setting default settings via env vars
+#
+# e.g. Setting the env var -Ecluster.name=testcluster
+#
+# will cause Elasticsearch to be invoked with -Ecluster.name=testcluster
+#
+# see https://www.elastic.co/guide/en/elasticsearch/reference/5.0/settings.html#_setting_default_settings
+
+es_opts=''
+
+while IFS='='  read -r envvar_key envvar_value
+do
+    case $envvar_key in
+    [a-z.]*)
```

Comment:
At first I was really confused here, because Bash can't cope with dots in variable names. After thinking it through a bit, I realised that you are never actually asking Bash to handle dotted variable names _as such_...  very cunning.

From `man bash`:
> name: A word consisting only of alphanumeric characters and underscores, and beginning with an alphabetic character or an underscore.  Also referred to as an identifier.

-

Beware of [accidental case-insensitive matching](http://stackoverflow.com/questions/10695029/why-isnt-the-case-statement-case-sensitive-when-nocasematch-is-off) here. On my workstation, this match does not give the desired behaviour, and _all_ environment variables are passed to ES.

```
$ case ABC in [a-z]*) echo true; esac
true
```
Being in a container obviously helps here, but yeah, shell is a wild language. I guess the acceptance tests will notice pretty quick if this somehow flips in the image later.